src/nvim/ex_getln.c

```diff
@@ -2325,7 +2310,7 @@ void putcmdline(int c, int shift)
       draw_cmdline(ccline.cmdpos, ccline.cmdlen - ccline.cmdpos);
     msg_no_more = FALSE;
   } else {
-    ui_ext_cmdline_char(c, shift);
+    ui_call_cmdline_char(cstr_to_string((char *)(&c)), shift);
```

Comment:
`cchar_to_string((char)c)` would be better, maybe (add to `api/private/helpers.c`)
```
/// Allocates a String consisting of a single char. Does not support multibyte
/// characters. The resulting string is also NUL-terminated, to facilitate
/// interoperating with code using C strings.
///
/// @param char the char to convert
/// @return the resulting String, if the input char was NUL, an
///         empty String is returned
String cchar_to_string(char c)
{
    char buf[] = {c, NUL};
    return (String) {
        .data = xmemdupz(buf, 1),
        .size = (c != NUL) ? 1 : 0
    };
}
```